.github/workflows/wheels.yml

```diff
@@ -96,10 +157,57 @@ jobs:
       - uses: actions/upload-artifact@v3
         with:
           path: ./wheelhouse/*.whl
-      - name: Install twine
-        run: python -m pip install twine
-      - name: Upload to PyPI
-        run: twine upload ./wheelhouse/*.whl
-        env:
-          TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
-          TWINE_USERNAME: qiskit
+      - name: Publish package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          packages-dir: wheelhouse/
+  sdist:
+    name: Build and publish terra sdist
+    runs-on: ${{ matrix.os }}
+    needs: ["build_wheels"]
+    environment: release
+    permissions:
+      id-token: write
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        name: Install Python
+        with:
+          python-version: '3.10'
+      - name: Install deps
+        run: pip install -U twine setuptools-rust wheel build
+      - name: Build sdist
+        run: python -m build . --sdist
+      - name: Publish package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+  metapackage:
+    name: Build and publish terra sdist
+    runs-on: ${{ matrix.os }}
+    needs: ["sdist"]
+    environment: release
+    permissions:
+      id-token: write
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        name: Install Python
+        with:
+          python-version: '3.10'
+      - name: Install deps
+        run: pip install -U twine setuptools-rust wheel build
+      - name: Build sdist
+        run: |
+          cd qiskit_pkg
+          python -m build .
```

Comment:
Minor nitpicking:
- the step name is "build sdist" but this builds an sdist and a wheel
- the run component possibly either wants a `set -e`, or alternatively skip the `cd` and use the `working-directory` YAML key?

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/10999/commits/04043a8c241e542b2dece33c9ecc6c9e053b7c01